graql/executor/ComputeExecutorImpl.java

```diff
@@ -126,31 +136,36 @@
         }
     }
 
-    Stream<ConceptList> stream(GraqlCompute.Path query) {
+    @Override
+    public Stream<ConceptList> stream(GraqlCompute.Path query) {
         return runComputePath(query);
     }
 
-    Stream<ConceptSetMeasure> stream(GraqlCompute.Centrality query) {
+    @Override
+    public Stream<ConceptSetMeasure> stream(GraqlCompute.Centrality query) {
         return runComputeCentrality(query);
     }
 
-    Stream<ConceptSet> stream(GraqlCompute.Cluster query) {
+    @Override
+    public Stream<ConceptSet> stream(GraqlCompute.Cluster query) {
         return runComputeCluster(query);
     }
 
+    @Override
     public final ComputerResult compute(@Nullable VertexProgram<?> program,
                                         @Nullable MapReduce<?, ?, ?, ?, ?> mapReduce,
                                         @Nullable Set<LabelId> scope,
                                         Boolean includesRolePlayerEdges) {
 
-        return tx.session().transactionOLAP().compute(program, mapReduce, scope, includesRolePlayerEdges);
+        return new OLAP(hadoopGraph).compute(program, mapReduce, scope, includesRolePlayerEdges);
     }
 
+    @Override
     public final ComputerResult compute(@Nullable VertexProgram<?> program,
                                         @Nullable MapReduce<?, ?, ?, ?, ?> mapReduce,
                                         @Nullable Set<LabelId> scope) {
 
-        return tx.session().transactionOLAP().compute(program, mapReduce, scope);
+        return new OLAP(hadoopGraph).compute(program, mapReduce, scope);
```

Comment:
is OLAP the new name of TransactionOLAP?

Reply:
or OLAP operation?

Reply:
has been renamed to OLAPOperation for now, because we can run many of these within one Transaction still